### PR TITLE
Add documentation related to the support for payment apps

### PR DESCRIPTION
**Pull request by korycins**
The docs that cover the feature explained here: https://github.com/saleor/saleor/issues/11258

Fixes https://github.com/saleor/saleor/issues/12306

### CODE REVIEW

File: docs/developer/payments.mdx [hunk 1]
```diff
@@ -516,7 +818,113 @@ for `transaction.canceledAmount` will not take into account the amount from `CAN
...
+| :---: | :-------------------: | :----------: | :-----------------------: | :----: | :-------------------------: | :-------------------------------: | :----------------------------: |
+|   1   | AUTHORIZATION_SUCCESS |     AB12     | 2022-03-28T12:50:33+00:00 |   10   |              0              |                 0                 |               10               |
+|   2   |    CHARGE_SUCCESS     |     YZ13     | 2022-03-28T12:51:33+00:00 |   3    |              3              |                 0                 |               7                |
+
+The `AUTHORIZATION_SUCCESS` event is attached to the transaction, and its amount is added to `transaction.authorizedAmount`. Similarly, the `CHARGE_SUCCESS` event is attached to the transaction,
+and its amount is added to `transaction.chargedAmount`. Finally, the amount of the `AUTHORIZATION_SUCCESS` event is subtracted from the `transaction.authorizedAmount`.
+
+---
 
 ## Payment Gateway
```
Comment (mmiszy) [inline docs/developer/payments.mdx:930]: "Using PaymentGateway is strictly related to the checkout object" – is not also related to Order object now?

File: docs/developer/payments.mdx [hunk 2]
```diff
@@ -8,12 +8,301 @@ The process below describes the key milestones in the Saleor payment flow.
...
+- `AUTHORIZATION_ACTION_REQUIRED` or `CHARGE_ACTION_REQUIRED` indicates that additional action is required from the customer to
+  finalize the payment. Calling [transactionProcess](api-reference/payments/mutations/transaction-process.mdx) will be required to pass the result of the action to the app.
+- `AUTHORIZATION_SUCCESS` indicates that funds have been authorized.
+- `AUTHORIZATION_FAILURE` indicates that authorization has failed.
+- `AUTHORIZATION_REQUEST` indicates that authorization is in a pending state.
+- `CHARGE_SUCCESS` indicates that funds have been charged.
+- `CHARGE_FAILURE` indicates that charging has failed.
+- `CHARGE_REQUEST` indicates that charging is in a pending state.
+
+## Saleor App with existing Checkout App
```
Comment (mmiszy) [inline docs/developer/payments.mdx:305]: The part about `transactionEventReport` from this section also applies to Payment Apps. We use `transactionEventReport` to react to Adyen webhooks for example.

File: docs/developer/payments.mdx [hunk 3]
```diff
@@ -8,12 +8,301 @@ The process below describes the key milestones in the Saleor payment flow.
 Additional steps may also occur along the way; however, the purpose of this instruction is to
 deliver a base reference for the user to work with.
 
-Saleor distinguishes two different ways of processing a payment:
+Saleor distinguishes three different ways of processing a payment:
 
 - By using a [plugin](extending/overview#plugins) embedded in the Saleor code.
-- By using a [Saleor App](extending/overview#apps-recommended) (recommended).
+- By using a [Saleor Payment App](extending/overview#apps-recommended)
+- By using a [Saleor App with existing Checkout App](extending/overview#apps-recommended).
```
Comment (mmiszy) [inline docs/developer/payments.mdx:15]: These two links are the same

**Review by timuric — APPROVED** [review]

File: docs/developer/payments.mdx (hunk 3 above)
Comment (korycins) [reply docs/developer/payments.mdx:15]: Yeap. They points to general concept of the `App`. Do you think that I should link something different?

File: docs/developer/payments.mdx (hunk 1 above)
Comment (korycins) [reply docs/developer/payments.mdx:930]: No. PaymentPlugin can be used only with Checkout object. Probably I should rename this from `Payment Gateway` to Payment Plugin`

File: docs/developer/payments.mdx (hunk 2 above)
Comment (korycins) [reply docs/developer/payments.mdx:305]: right. I need to rebuild this page. Thanks!

File: docs/developer/payments.mdx (hunk 3 above)
Comment (korycins) [reply docs/developer/payments.mdx:15]: Nevermind. I already changed this structure

**Review by mmiszy — APPROVED** [review]

**Pull request merged**